From a43bb39b5c710e79e921fb76341bacc418cfde78 Mon Sep 17 00:00:00 2001
From: Qu Wenruo <quwenruo@cn.fujitsu.com>
Date: Wed, 8 Oct 2014 10:19:08 +0800
Subject: btrfs: Fix compile error when CONFIG_SECURITY is not set.

Fix the following compile error when CONFIG_SECURITY is not set:

error: 'struct security_mnt_opts' has no member named 'num_mnt_opts'

Signed-off-by: Qu Wenruo <quwenruo@cn.fujitsu.com>
Signed-off-by: Chris Mason <clm@fb.com>

diff --git a/fs/btrfs/super.c b/fs/btrfs/super.c
index 2250284..a2b97ef 100644
--- a/fs/btrfs/super.c
+++ b/fs/btrfs/super.c
@@ -1248,6 +1248,7 @@ static int setup_security_options(struct btrfs_fs_info *fs_info,
 	if (ret)
 		return ret;
 
+#ifdef CONFIG_SECURITY
 	if (!fs_info->security_opts.num_mnt_opts) {
 		/* first time security setup, copy sec_opts to fs_info */
 		memcpy(&fs_info->security_opts, sec_opts, sizeof(*sec_opts));
@@ -1260,6 +1261,7 @@ static int setup_security_options(struct btrfs_fs_info *fs_info,
 		 */
 		security_free_mnt_opts(sec_opts);
 	}
+#endif
 	return ret;
 }
 
-- 
cgit v0.10.1

